crates/matrix-sdk-sqlite/src/lib.rs

```diff
@@ -35,3 +38,53 @@ pub use self::state_store::SqliteStateStore;
 
 #[cfg(test)]
 matrix_sdk_test::init_tracing_for_tests!();
+
+/// A configuration structure used for opening a store.
+pub struct StoreOpenConfig {
+    /// Path to the database, without the file name.
+    path: PathBuf,
+    /// Passphrase to open the store, if any.
+    passphrase: Option<String>,
+    /// The pool configuration for [`deadpool_sqlite`].
+    pool_config: PoolConfig,
+}
+
+impl StoreOpenConfig {
+    /// Create a new [`StoreOpenConfig`] with a path representing the directory
+    /// containing the store database, and its passphrase if encoded.
+    pub fn new<P>(path: P, passphrase: Option<&str>) -> Self
```

Comment:
If we're creating a builder, perhaps it's better to move the optional `passphrase` parameter into its own setter.